apps/browser-extension-wallet/src/hooks/useBuildDelegation.ts

```diff
@@ -1,15 +1,45 @@
+import { useEffect } from 'react';
+import { useStakePoolDetails } from '@src/features/stake-pool-details/store';
+import { StakingError } from '@src/views/browser-view/features/staking/types';
 import { useWalletStore } from '../stores';
-import { useCallback } from 'react';
-import { CardanoTxBuild } from '../types';
 import { useDelegationStore } from '../features/delegation/stores';
-import { Wallet } from '@lace/cardano';
+import { InputSelectionError, InputSelectionFailure } from '@cardano-sdk/input-selection';
 
-export const useBuildDelegation = (): (() => Promise<CardanoTxBuild>) => {
+const ERROR_MESSAGES: { [key: string]: StakingError } = {
+  [InputSelectionFailure.UtxoFullyDepleted]: StakingError.UTXO_FULLY_DEPLETED,
+  [InputSelectionFailure.UtxoBalanceInsufficient]: StakingError.UTXO_BALANCE_INSUFFICIENT
+};
+
+export const useBuildDelegation = (): void => {
   const { inMemoryWallet } = useWalletStore();
-  const { selectedStakePool } = useDelegationStore();
+  const { selectedStakePool, setDelegationTxBuilder, setDelegationTxFee } = useDelegationStore();
+  const { setIsBuildingTx, setStakingError } = useStakePoolDetails();
+
+  useEffect(() => {
+    const buildDelegation = async () => {
+      try {
+        setIsBuildingTx(true);
+        const txBuilder = inMemoryWallet.createTxBuilder();
+        const tx = await txBuilder.delegate(selectedStakePool.id).build().inspect();
+        setDelegationTxBuilder(txBuilder);
+        setDelegationTxFee(tx.body.fee.toString());
+        setStakingError();
+      } catch (error) {
+        if (error instanceof InputSelectionError) {
```

Comment:
`instanceof` won't work for errors until we do LW-6749
```suggestion
        if (typeof error === 'object' && Object.values(InputSelectionFailure).includes((error as any).failure)) {
```